### PR TITLE
Fixes a bunch of hotkey preferences bugs

### LINKED ISSUE #56382: New keybinds shriek that they need updating, even if they are not automatically bound to anything.
image

### LINKED ISSUE #59491: Keybinding conflict system is fucking useless
1. Join ~~terry~~ any server.
2. Keybind conflict message!
3. Reset keybinds to default.
4. Hit save on preferences
5. Reconnect.
6. Keybind conflict message!

### PR DESCRIPTION
**Pull request by Gamer025**
## About The Pull Request
This fixes the problem of the keybind conflict message being shown over and over again until you manually save your preferences with correct keybinds + resetting to classic keys works now
You had to manually save your prefs because the code didn't save the new unbound keys to the prefs file
Classic mode was broken because the emote hotkeys didn't have classic_keys set to Unbound by default

New hotkeys are now actually set to Unbound if no default key is set
This was broken for emote hotkeys because classic_keys = list("Unbound") was missing for them
It was also broken because the code assumed conflicting keys if some Unbound key already existed ...
The code also used classic key defaults even if you had hotkey mode enabled thats fixed now too

Fixes #59491 and fixes #56382
Closes #59490

## Why It's Good For The Game
I guess constantly getting a message to fix your keybinds is somewhat annoying but I honstly just don't want random "KEYBINDING CONFLICT IS STILL A THING" runtimes in the logs

## Changelog
:cl:
fix: The message for new duplicate keybinds should only appear once now
fix: Resetting your keybinds to classic mode works correctly now
fix: Hotkey mode is now actually used when new hotkeys with a default setting are added
fix: You will no longer get a duplicate keybinding conflict message for new hotkeys with no default key, they will just be silently set to unbound
/:cl:

### CODE REVIEW

File: code/modules/client/preferences_savefile.dm [hunk 1]
```diff
@@ -127,11 +127,12 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 
 /datum/preferences/proc/announce_conflict(list/notadded)
 	to_chat(parent, "<span class='warningplain'><b><u>Keybinding Conflict</u></b></span>\n\
-					<span class='warningplain'><b>There are new <a href='?_src_=prefs;preference=tab;tab=3'>keybindings</a> that default to keys you've already bound. The new ones will be unbound.</b></span>")
+					<span class='warningplain'><b>There are new <a href='?_src_=prefs;preference=tab;tab=3'>keybindings</a> that default to keys you've already bound or that need to be manualy set. The new ones will be unbound.</b></span>")
```
Comment (tralezab) [inline code/modules/client/preferences_savefile.dm:130]:
```suggestion
					<span class='warningplain'><b>There are new <a href='?_src_=prefs;preference=tab;tab=3'>keybindings</a> that default to keys you've already bound or that need to be manually set. The new ones will be unbound.</b></span>")
```

**Comment by LemonInTheDark** [comment]
He won

**Comment by cacogen** [comment]
Should announce_conflict() even be triggered by keybindings that need to be manually set? Doesn't that mean they're unbound by default?

**Comment by Gamer025** [comment]
> Should announce_conflict() even be triggered by keybindings that need to be manually set? Doesn't that mean they're unbound by default?

I think you are right and the code actually should be setting new keys to unbound (or their default value if provided) and only alert on conflict.
However since all the emote hotkeys were missing classic_keys = list("Unbound") the check_keybindings code was broken and didn't set the new emotes to Unbound.
This however begs the question what the "hotkeys" variable does as both codepaths here are identical in my opinion (the only difference here is the different name for the temporary for var):
image
Im quite sure the first codeblock should be `var/hotkeytobind in kb.hotkey_keys`
The fact that the hotkeys var was added by a commit without PR with only "keybindings save file fix" as description isn't very helpful.
Also im quite sure check_keybindings() also has another bug where it will not assign new hotkeys to Unbound if any other hotkey is already Unbound because it thinks that's a conflict ...
Why is this code broken in so many ways

**Comment by Gamer025** [comment]
Ok all the issues should be fixed now, see the edited PR description / last commit message for what changed

**Pull request merged**